infrastructure/constructs/lambda_config.py

```diff
@@ -1,6 +1,15 @@
-from aws_cdk import Duration, aws_lambda
+from aws_cdk import Duration, aws_lambda, aws_logs
+
+from geostore.environment import is_production
 
 PYTHON_RUNTIME = aws_lambda.Runtime.PYTHON_3_9
 
 DEFAULT_LAMBDA_MAX_MEMORY_MEBIBYTES = 1024
 DEFAULT_LAMBDA_TIMEOUT = Duration.seconds(60)
+
+
+def log_retention_environment() -> aws_logs.RetentionDays:
+    if is_production():
```

Comment:
It shouldn't be possible for `is_production` to change return value during a deployment, making its result effectively constant at runtime. So we could pull this out as a constant:

```python
if is_production():
    RETENTION_DAYS = aws_logs.RetentionDays.ONE_YEAR
else:
    RETENTION_DAYS = aws_logs.RetentionDays.THREE_MONTHS
```